### PR TITLE
Test converter

### PR DESCRIPTION
**Pull request by viskey98**
This is the unittest for testing converter by checking if the hashed files created by converter actually exist or not.
Please suggest any changes if required :)

### CODE REVIEW

File: tests/test_converter.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        self.page_path = '/tmp/test/'
+        if not os.path.exists('/tmp/test'):
+            os.mkdir('/tmp/test')
+        if not os.path.exists('/tmp/test/depth'):
+            os.mkdir('/tmp/test/depth')
+        self.hname1 = ""
+        self.hname2 = ""
+        with open(os.path.join(self.page_path, 'index.html'),   'w') as f:
+            f.write(self.content)
+            f.close()
```
Comment (afeena) [inline tests/test_converter.py]: no need to use `close` with `with`

File: tests/test_converter.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with open(os.path.join(self.page_path, 'depth/page.html'), 'w') as f:
+            f.write(self.content)
+            f.close()
+        self.cnv = Converter()
+
+    def test_converter(self):
+        self.cnv.convert(self.page_path)
+        f = open(os.path.join(self.page_path, 'meta.json'), 'r')
```
Comment (afeena) [inline tests/test_converter.py]: it's better to use `with` while reading the file

**Comment by viskey98** [comment]
@afeena The suggested changes have been made. This test runs fine in my machine but travis sometimes passes and sometimes fails for the same code. Why does that occur?

File: tests/test_converter.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        with open(os.path.join(self.page_path, 'index.html'),   'w') as f:
+            f.write(self.content)
+        with open(os.path.join(self.page_path, 'depth/page.html'), 'w') as f:
+            f.write(self.content)
+        self.cnv = Converter()
+
+    def test_converter(self):
+        self.cnv.convert(self.page_path)
+        with open(os.path.join(self.page_path, 'meta.json'), 'r') as f:
+            s = f.read()
```
Comment (afeena) [inline tests/test_converter.py]: why don't you use `json.load` here?

File: tests/test_converter.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        index = s.index('"index.html"') + len('"index.html"') + 12
```
Comment (afeena) [inline tests/test_converter.py]: parse file as json and work with it

File: tests/test_converter.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        self.hname2 = ""
+        with open(os.path.join(self.page_path, 'index.html'),   'w') as f:
+            f.write(self.content)
+        with open(os.path.join(self.page_path, 'depth/page.html'), 'w') as f:
+            f.write(self.content)
+        self.cnv = Converter()
+
+    def test_converter(self):
+        self.cnv.convert(self.page_path)
+        with open(os.path.join(self.page_path, 'meta.json'), 'r') as f:
```
Comment (afeena) [inline tests/test_converter.py]: 'r' is default property, no need to specify

File: tests/test_converter.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            s = f.read()
+        index = s.index('"index.html"') + len('"index.html"') + 12
+        self.hname1 = s[index: index + 32]
+        index = s.index('"depth/page.html"') + len('"depth/page.html"') + 12
```
Comment (afeena) [inline tests/test_converter.py]: same here

File: tests/test_converter.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import shutil
+from converter import Converter
+
+
+class TestConverter(unittest.TestCase):
+
+    def setUp(self):
+        self.content = '<html><body></body></html>'
+        self.page_path = '/tmp/test/'
+        if not os.path.exists('/tmp/test'):
```
Comment (afeena) [inline tests/test_converter.py]: for creating multiple directories you can use `os.makedirs`

File: tests/test_converter.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        index = s.index('"index.html"') + len('"index.html"') + 12
+        self.hname1 = s[index: index + 32]
+        index = s.index('"depth/page.html"') + len('"depth/page.html"') + 12
+        self.hname2 = s[index: index + 32]
+        assert(os.path.exists(self.page_path + self.hname1) and
+               os.path.exists(self.page_path + self.hname2))
+
+    def tearDown(self):
+        shutil.rmtree('/tmp/test/depth')
+        shutil.rmtree('/tmp/test')
```
Comment (afeena) [inline tests/test_converter.py]: when you delete parent directory, others are also deleted. no need to delete it directly

**Comment by viskey98** [comment]
@afeena The suggested changes have been done, please look into it :)

File: tests/test_converter.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            f.write(self.content)
+        with open(os.path.join(self.page_path, 'depth/page.html'), 'w') as f:
+            f.write(self.content)
+        self.cnv = Converter()
+
+    def test_converter(self):
+        self.cnv.convert(self.page_path)
+        with open(os.path.join(self.page_path, 'meta.json')) as f:
+            s = f.read()
+            s = json.loads(s)
```
Comment (afeena) [inline tests/test_converter.py]: json can be loaded from buffer directly

**Comment by viskey98** [comment]
@afeena suggested change has been made :)

File: tests/test_converter.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        with open(os.path.join(self.page_path, 'index.html'), 'w') as f:
+            f.write(self.content)
+        with open(os.path.join(self.page_path, 'depth/page.html'), 'w') as f:
+            f.write(self.content)
+        self.cnv = Converter()
+
+    def test_converter(self):
+        self.cnv.convert(self.page_path)
+        with open(os.path.join(self.page_path, 'meta.json')) as f:
+            s = json.loads(f.read())
```
Comment (afeena) [inline tests/test_converter.py]: please use `json.load(f)` here :)

Json has two function: `load` and `loads`:
`load` can directly read from text stream
`loads` read json from a string

**Comment by viskey98** [comment]
As I was dealing with a string I got a little confused there whether to use `load()` or `loads()`. Now I have implemented with load directly from the buffer. Please look into it

**Pull request merged**